Source/DafnyLanguageServer/Plugins/QuickFixerHelpers.cs

```diff
@@ -0,0 +1,153 @@
+    var token = range.ToBoogieToken(text);
+    var startTokenPos = token.StartToken.pos;
+    var endTokenPos = token.EndToken.pos + token.EndToken.val.Length;
+    var length = endTokenPos - startTokenPos;
+    if (startTokenPos < 0 || endTokenPos < startTokenPos || endTokenPos >= text.Length) {
+      return ""; // Safeguard
+    }
+
+    return text.Substring(startTokenPos, length);
+  }
+
+  /// <summary>
+  /// Given the position of a closing brace '}' and the position of the opening brace '{',
+  /// returns spacing that can be used to insert a statement before it,
+  /// as well as spacing for after the statement
+  /// </summary>
+  /// <param name="endToken">The position of the closing brace</param>
+  /// <param name="text">The document text</param>
+  /// <returns>(extra indentation for a statement, current indentation)</returns>
+  public static (string, string) GetIndentationBefore(IToken endToken, IToken startToken, string text) {
+    var pos = endToken.pos + endToken.val.Length - 1;
+    var indentation = 0;
+    var indentationBrace = endToken.col - 1;
+    var firstNewline = true;
+    var useTabs = false;
+    // Look for the first newline
+    while (0 <= pos && pos < text.Length && (text[pos] != '\n' || firstNewline)) {
+      if (text[pos] == '\t') {
+        useTabs = true;
+      }
+
+      if (text[pos] == '\n') {
+        if (!firstNewline) {
+          break;
+        }
+
+        firstNewline = false;
+      }
+
+      if (!firstNewline) {
+        if (text[pos] == ' ' || text[pos] == '\t') {
+          indentation++;
+        } else {
+          indentation = 0;
+        }
+      }
+
+      pos--;
+    }
+
+    if (startToken.line == endToken.line - 1) {
+      // Override case {\n} with some spacing, we return a default value of 2 spaces or 1 tab, if we find some
+      indentation = indentationBrace + (useTabs ? 1 : 2);
+    } else if (startToken.line == endToken.line) {
+      // Override case { ... } on one line.
+      indentationBrace = startToken.col - 1;
+      indentation = indentationBrace + 1;
+    }
+
+    indentation = Math.Max(indentationBrace, indentation);
+
+    return (
+      new string(useTabs ? '\t' : ' ', indentation - indentationBrace),
+      new string(useTabs ? '\t' : ' ', indentationBrace));
+  }
+
+  /// <summary>
+  /// Given an opening brace (typically where an error is detected),
+  /// finds and returns the closing brace token.
+  /// </summary>
+  /// <param name="program">The program</param>
+  /// <param name="documentUri">The document URI</param>
+  /// <param name="openingBrace">A token whose pos is the absolute position of the opening brace</param>
+  /// <returns>The token of a matching closing brace, typically the `ÈndTok` of a BlockStmt</returns>
+  public static IToken? GetMatchingEndToken(Dafny.Program program, string documentUri, IToken openingBrace) {
+    // Look in methods for BlockStmt with the IToken as opening brace
```

Comment:
Could the brace not also be for a class or module body? If methods are the only use case, could this be documented?